### PR TITLE
chore: cleanup nft, addressbook and marketplace module

### LINKED ISSUE #97: Remove NFT module in dev branch
Relevent module: https://github.com/CudoVentures/cudos-node/tree/dev/x/nft

### LINKED ISSUE #98: Remove marketplace module in dev branch
relevent module: https://github.com/CudoVentures/cudos-node/tree/dev/x/marketplace

### LINKED ISSUE #99: Remove addrbook module

### PR DESCRIPTION
**Pull request by catconcat**
## Description
This PR removes nft, addressbook and marketplace modules. 

Closes #97, closes #98 and closes #99

### CODE REVIEW

**Comment by nghuyenthevinh2000** [comment]
LGTM!

File: app/app_upgrades.go [hunk 1]
```diff
@@ -69,7 +70,7 @@ func setHandlerForVersion_1_1(app *App) {
 
 	if upgradeInfo.Name == upgradeVersion && !app.UpgradeKeeper.IsSkipHeight(upgradeInfo.Height) {
 		storeUpgrades := storetypes.StoreUpgrades{
-			Added: []string{authz.ModuleName, group.ModuleName, addressbookTypes.ModuleName, marketplaceTypes.ModuleName},
+			Added: []string{authz.ModuleName, group.ModuleName, AddressBookModuleName, MarketplaceModuleName},
```
Comment (jgtormo) [inline app/app_upgrades.go:73]: we're currently running node version 1.1  shouldn't these changes be applied to a `setHandlerForVersion_1_2` function instead for the new upgrade?

File: x/nft/client/cli/cli_test.go [hunk 2]
```diff
@@ -1,397 +0,0 @@
-package cli_test
-
-import (
-	"fmt"
-	"testing"
-
-	"github.com/gogo/protobuf/proto"
-	"github.com/stretchr/testify/suite"
-	"github.com/tidwall/gjson"
```
Comment (jgtormo) [inline x/nft/client/cli/cli_test.go:9]: I think that's the only place where a library from `tidwall` was used, so could remove that from app.go and app.mod
There seem to be some other unused libraries as well so a cleanup would be good.

File: app/app_upgrades.go (hunk 1 above)
Comment (catconcat) [reply app/app_upgrades.go:73]: We've removed the 3 modules from the codebase and thus making their `ModuleName` inaccessible, so these changes are solely intended to making the code compiles, and not related to the upgrade process.

Comment (jgtormo) [reply app/app_upgrades.go:73]: yes, apologies let me rephrase. I had understood that these functions were used also as a record, and something that is used/referenced when you build a specific version. So for example Osmosis has a slightly different structure but has a different file and folder for each version: https://github.com/osmosis-labs/osmosis/tree/main/app/upgrades

Shouldn't we leave the current version as it is, and start including this in a new version? Rather than modifying the function of a version of the code that's already released and running.

Comment (jgtormo) [reply app/app_upgrades.go:73]: related to this message as well: https://cudocrypto.slack.com/archives/C05RC14GBEW/p1703155003933409?thread_ts=1703154904.665749&cid=C05RC14GBEW

But please let me know if what I'm saying doesn't make sense.

Comment (catconcat) [reply app/app_upgrades.go:73]: @jgtormo 
> Shouldn't we leave the current version as it is, and start including this in a new version?

Yes, that is essentially what I am doing. My changes are similar to this of Osmosis:  https://github.com/osmosis-labs/osmosis/blob/ec3603396e022403dfa7e7765f819c5d304d3003/app/upgrades/v9/constants.go#L16-L26

> related to this message as well: https://cudocrypto.slack.com/archives/C05RC14GBEW/p1703155003933409?thread_ts=1703154904.665749&cid=C05RC14GBEW

I can't access the link, it requires a Cudos email.

Comment (jgtormo) [reply app/app_upgrades.go:73]: isn't [this line](https://github.com/osmosis-labs/osmosis/blob/ec3603396e022403dfa7e7765f819c5d304d3003/app/upgrades/v9/constants.go#L25) explicitly signalling that in the upgrade from v8 to v9 the `claim` module will be removed? Rather than modifying to v8 file and removing its reference from there

Comment (nghuyenthevinh2000) [reply app/app_upgrades.go:73]: @jgtormo I think there is a misunderstanding here, these changes are only to make the cleanup compatible with old code base.

So, when @tropicaldog removed those three modules, it created **syntax errors** since there was no longer those module names. He had to add those names in to fix syntax errors. It is not related to upgrades.

Or, perhaps, what you mean is to create a separate file for upgrade constants? Sorry if there are any misunderstandings

Comment (jgtormo) [reply app/app_upgrades.go:73]: yes, and I apologise as well if I'm not being clear enough or what I say just doesn't make sense. What I'm trying to say is that I was thinking the cleanup would be done as part of an upgrade, and that everything that references the currently running code with these modules in would stay unaffected. Mainnet is currently running v1.1 and [`setHandlerForVersion1_1`](https://github.com/CudoVentures/cudos-node/blob/12a356cca43f15e89d7b883302f3dba3b6e58fcd/app/app_upgrades.go#L43C23-L43C23) is, as I understand, specific to that version/upgrade to that version. So rather than modifying code that specifically relates to a release that all validators are currently running on mainnet, what I understood this file to be and what I seem to understand Osmosis do is signal in a new upgrade that some modules are being deleted.

In other words, shouldn't removing these three modules create syntax errors, and shouldn't that be the expected behaviour? Because in the v1.1 release they are there, and this function relates to v1.1 so if they are gone that should be an issue if trying to run that version.

Comment (jgtormo) [reply app/app_upgrades.go:73]: had a chat with @nbaum and he has some different comments, but closing this for now and merging.

**Review by jgtormo — APPROVED** [review]
